core/src/OpenMP/Kokkos_OpenMP_Instance.hpp

```diff
@@ -134,28 +142,24 @@ inline bool OpenMP::impl_is_initialized() noexcept {
   return Impl::OpenMPInternal::singleton().is_initialized();
 }
 
-inline bool OpenMP::in_parallel(OpenMP const&) noexcept {
-  // FIXME_OPENMP We are forced to use t_openmp_instance because the function is
-  // static and does not use the OpenMP object
-  return ((Impl::OpenMPInternal::singleton().m_level < omp_get_level()) &&
-          (!Impl::t_openmp_instance ||
-           Impl::t_openmp_instance->m_level < omp_get_level()));
+inline bool OpenMP::in_parallel(OpenMP const& exec_space) noexcept {
+  return (
+      (exec_space.impl_internal_space_instance()->m_level < omp_get_level()) &&
+      (!Impl::t_openmp_instance ||
+       Impl::t_openmp_instance->m_level < omp_get_level()));
 }
 
-inline int OpenMP::impl_thread_pool_size() noexcept {
-  // FIXME_OPENMP We are forced to use t_openmp_instance because the function is
-  // static
-  return OpenMP::in_parallel()
+inline int OpenMP::impl_thread_pool_size(OpenMP const& exec_space) noexcept {
+  return OpenMP::in_parallel(exec_space)
              ? omp_get_num_threads()
              : (Impl::t_openmp_instance
                     ? Impl::t_openmp_instance->m_pool_size
-                    : Impl::OpenMPInternal::singleton().m_pool_size);
+                    : exec_space.impl_internal_space_instance()->m_pool_size);
 }
 
-KOKKOS_INLINE_FUNCTION
-int OpenMP::impl_thread_pool_rank() noexcept {
-  // FIXME_OPENMP We are forced to use t_openmp_instance because the function is
-  // static
+inline int OpenMP::impl_thread_pool_rank() noexcept {
```

Comment:
Did you mean to drop the `KOKKOS_INLINE_FUNCTION` annotation?  If so we probably should drop the `KOKKOS_IF_ON_{HOST,DEVICE}`